include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -24,10 +24,228 @@
 #    include "hetero/algorithm_impl_hetero.h" //TODO: for __brick_copy
 #endif
 
+#if _ONEDPL_CPP20_RANGES_PRESENT
+#    include <ranges>
+#    include <functional>
+#    include "algorithm_ranges_impl.h"
+#endif
+
 namespace oneapi
 {
 namespace dpl
 {
+namespace ext
+{
+
+#if _ONEDPL_CPP20_RANGES_PRESENT
+namespace ranges
+{
+
+// [alg.foreach]
+
+struct for_each_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             std::indirectly_unary_invocable<std::projected<std::ranges::iterator_t<_R>, _Proj>>  _Fun>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+    constexpr auto
```

Comment:
Why do we make them `constexpr`? It's not a part of C++ standard for parallel algorithms.

Reply:
Why `constexpr` is not a part of C++?
For serial  - is a part:
```
constexpr for_each_result<[ranges::borrowed_iterator_t](http://en.cppreference.com/w/cpp/ranges/borrowed_iterator_t)<R>, Fun>
    for_each( R&& r, Fun f, Proj proj = {} );
```

For parallel - I don't see any ban for that:

```
2.11. constexpr parallel range algorithms
[[P2902R0]](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3179r1.html#biblio-p2902r0) suggests allowing algorithms with execution policies to be used in constant expressions. We do not consider that as a primary design goal for our work, however we will happily align with that proposal in the future once it progresses towards adoption into the working draft.
```

Reply:
Discussed offline, agreed to remove  `constexpr`

Reply:
done